src/Voron/Data/BTrees/TreeMutableState.cs

```diff
@@ -1,39 +1,84 @@
 ﻿using System;
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 using Voron.Global;
 using Voron.Impl;
 
 namespace Voron.Data.BTrees
 {
-    public sealed unsafe class TreeMutableState
+
+    [StructLayout(LayoutKind.Explicit, Size=64)]
+    public unsafe struct TreeMutableState(LowLevelTransaction tx)
```

Comment:
I don't think that this should be 1:1 memory aligned as the persistent struct.
This is meant to allow far greater freedom, and isn't persisted, after all.
Better to let C# to just order this as needed. For example, all the longs aren't 8 bytes aligned, etc.

Reply:
The idea is to allow the JIT to emit vector instructions instead. If we do field by field, then for example a copy would require several sequences of mov to register, mov to location for each field.

Reply:
Sure, but that happens _once_ per transaction commit.
Accessing those fields is a lot more common, and happens a lot more frequently

Reply:
https://github.com/ravendb/ravendb/pull/17981#discussion_r1448691298